### ISSUE TITLE
报错了OpenSSL 3 is not yet supported. Please use OpenSSL 1.x

### ISSUE THREAD

**Issue by Zhanghao0150** [miaoshouai/miaoshouai-assistant#36]
$ openssl version
OpenSSL 1.0.2 22 Jan 2015

2023-08-07 13:53:14,368 - msai_file_downloader.py [line:205] - INFO: Download Attempt 1
2023-08-07 13:53:14,368 - msai_file_downloader.py [line:107] - INFO: Download error: OpenSSL 3 is not yet supported. Please use OpenSSL 1.x
2023-08-07 13:53:15,202 - msai_file_downloader.py [line:205] - INFO: Download Attempt 2
2023-08-07 13:53:15,202 - msai_file_downloader.py [line:107] - INFO: Download error: OpenSSL 3 is not yet supported. Please use OpenSSL 1.x
2023-08-07 13:53:15,369 - msai_file_downloader.py [line:205] - INFO: Download Attempt 2
2023-08-07 13:53:15,369 - msai_file_downloader.py [line:107] - INFO: Download error: OpenSSL 3 is not yet supported. Please use OpenSSL 1.x
2023-08-07 13:53:16,203 - msai_file_downloader.py [line:205] - INFO: Download Attempt 3
2023-08-07 13:53:16,203 - msai_file_downloader.py [line:107] - INFO: Download error: OpenSSL 3 is not yet supported. Please use OpenSSL 1.x
2023-08-07 13:53:16,370 - msai_file_downloader.py [line:205] - INFO: Download Attempt 3
2023-08-07 13:53:16,371 - msai_file_downloader.py [line:107] - INFO: Download error: OpenSSL 3 is not yet supported. Please use OpenSSL 1.x
2023-08-07 13:53:17,204 - msai_file_downloader.py [line:205] - INFO: Download Attempt 4
2023-08-07 13:53:17,204 - msai_file_downloader.py [line:107] - INFO: Download error: OpenSSL 3 is not yet supported. Please use OpenSSL 1.x
2023-08-07 13:53:17,371 - msai_file_downloader.py [line:205] - INFO: Download Attempt 4
2023-08-07 13:53:17,371 - msai_file_downloader.py [line:107] - INFO: Download error: OpenSSL 3 is not yet supported. Please use OpenSSL 1.x
2023-08-07 13:53:18,205 - msai_file_downloader.py [line:205] - INFO: Download Attempt 5
2023-08-07 13:53:18,205 - msai_file_downloader.py [line:107] - INFO: Download error: OpenSSL 3 is not yet supported. Please use OpenSSL 1.x
2023-08-07 13:53:18,372 - msai_file_downloader.py [line:205] - INFO: Download Attempt 5
2023-08-07 13:53:18,372 - msai_file_downloader.py [line:107] - INFO: Download error: OpenSSL 3 is not yet supported. Please use OpenSSL 1.x
2023-08-07 13:53:19,206 - msai_file_downloader.py [line:229] - INFO: https://image.civitai.com/xG1nkqKTMzGDvpLrqFT7WA/84735137-e96a-4d8b-9db1-96f44c1f8b00/width=450/173273.jpeg [  FAILED  ]
2023-08-07 13:53:19,665 - msai_file_downloader.py [line:229] - INFO: https://civitai.com/api/download/models/11523 [  FAILED  ]

pip的模块

$ /data/stable-diffusion-webui/venv/bin/python3 -m pip list
Package                   Version
------------------------- ------------------
absl-py                   1.4.0
accelerate                0.18.0
addict                    2.4.0
aenum                     3.1.15
aiofiles                  23.1.0
aiohttp                   3.8.5
aiosignal                 1.3.1
albumentations            1.3.1
altair                    5.0.1
antlr4-python3-runtime    4.9.3
anyio                     3.7.1
async-timeout             4.0.2
attrs                     23.1.0
audioread                 3.0.0
av                        10.0.0
basicsr                   1.4.2
beautifulsoup4            4.12.2
blendmodes                2022
boltons                   23.0.0
Brotli                    1.0.9
cachetools                5.3.1
certifi                   2023.7.22
cffi                      1.15.1
charset-normalizer        3.2.0
clean-fid                 0.1.35
click                     8.1.6
clip                      1.0
cmake                     3.27.0
coloredlogs               15.0.1
contourpy                 1.1.0
cryptography              41.0.3
cssselect2                0.7.0
cycler                    0.11.0
Cython                    3.0.0
dataclasses-json          0.5.14
decorator                 5.1.1
deprecation               2.1.0
duckduckgo-search         3.7.1
easydict                  1.10
einops                    0.4.1
exceptiongroup            1.1.2
face-alignment            1.3.5
facexlib                  0.3.0
fastapi                   0.94.0
ffmpy                     0.3.1
filelock                  3.12.2
filterpy                  1.4.5
flatbuffers               23.5.26
fonttools                 4.41.1
frozenlist                1.4.0
fsspec                    2023.6.0
ftfy                      6.1.1
future                    0.18.3
fvcore                    0.1.5.post20221221
gdown                     4.7.1
gfpgan                    1.3.8
gitdb                     4.0.10
GitPython                 3.1.30
google-auth               2.22.0
google-auth-oauthlib      1.0.0
gpt-index                 0.4.24
gradio                    3.32.0
gradio_client             0.2.7
greenlet                  2.0.2
grpcio                    1.56.2
h11                       0.12.0
h2                        4.1.0
hpack                     4.0.0
httpcore                  0.15.0
httpx                     0.24.1
huggingface-hub           0.16.4
humanfriendly             10.0
hyperframe                6.0.1
idna                      3.4
ifnude                    0.0.3
imageio                   2.31.1
imageio-ffmpeg            0.4.7
importlib-metadata        6.8.0
inflection                0.5.1
insightface               0.7.3
iopath                    0.1.9
Jinja2                    3.1.2
joblib                    1.3.1
jsonmerge                 1.8.0
jsonschema                4.18.4
jsonschema-specifications 2023.7.1
kiwisolver                1.4.4
kornia                    0.6.7
langchain                 0.0.132
langsmith                 0.0.18
lark                      1.1.2
lazy_loader               0.3
librosa                   0.8.0
lightning-utilities       0.9.0
linkify-it-py             2.0.2
lit                       16.0.6
llvmlite                  0.40.1
lmdb                      1.4.1
lpips                     0.1.4
lxml                      4.9.3
Markdown                  3.4.4
markdown-it-py            2.2.0
MarkupSafe                2.1.3
marshmallow               3.20.1
matplotlib                3.7.2
mdit-py-plugins           0.3.3
mdurl                     0.1.2
mediapipe                 0.10.2
mpmath                    1.3.0
multidict                 6.0.4
mypy-extensions           1.0.0
networkx                  3.1
numba                     0.57.1
numexpr                   2.8.4
numpy                     1.23.5
oauthlib                  3.2.2
omegaconf                 2.2.3
onnx                      1.14.0
onnxruntime               1.15.0
open-clip-torch           2.20.0
openai                    0.27.8
openapi-schema-pydantic   1.2.4
opencv-contrib-python     4.8.0.74
opencv-python             4.7.0.72
opencv-python-headless    4.8.0.74
orjson                    3.9.2
packaging                 23.1
pandas                    2.0.3
piexif                    1.1.3
Pillow                    9.5.0
pip                       22.0.2
platformdirs              3.9.1
pooch                     1.7.0
portalocker               2.7.0
prettytable               3.8.0
protobuf                  4.23.4
psutil                    5.9.5
pyasn1                    0.5.0
pyasn1-modules            0.3.0
pycparser                 2.21
pydantic                  1.10.12
pydub                     0.25.1
Pygments                  2.15.1
PyMatting                 1.1.8
pyOpenSSL                 20.0.0
pyparsing                 3.0.9
PySocks                   1.7.1
python-dateutil           2.8.2
python-multipart          0.0.6
pytorch-lightning         1.9.4
pytz                      2023.3
PyWavelets                1.4.1
PyYAML                    6.0.1
qudida                    0.0.4
realesrgan                0.3.0
referencing               0.30.0
regex                     2023.6.3
rehash                    1.0.1
rembg                     2.0.38
reportlab                 4.0.4
requests                  2.31.0
requests-oauthlib         1.3.1
resampy                   0.4.2
resize-right              0.0.2
rpds-py                   0.9.2
rsa                       4.9
safetensors               0.3.1
scikit-image              0.20.0
scikit-learn              1.3.0
scipy                     1.11.1
segment-anything          1.0
semantic-version          2.10.0
sentencepiece             0.1.99
setuptools                59.6.0
six                       1.16.0
smmap                     5.0.0
sniffio                   1.3.0
socksio                   1.0.0
sounddevice               0.4.6
soundfile                 0.12.1
soupsieve                 2.4.1
SQLAlchemy                1.4.49
starlette                 0.26.1
supervision               0.12.0
svglib                    1.5.1
sympy                     1.12
tabulate                  0.9.0
tb-nightly                2.14.0a20230522
tenacity                  8.2.2
tensorboard-data-server   0.7.1
termcolor                 2.3.0
threadpoolctl             3.2.0
tifffile                  2023.7.18
tiktoken                  0.4.0
timm                      0.6.7
tinycss2                  1.2.1
tokenizers                0.13.3
tomesd                    0.1.2
tomli                     2.0.1
toolz                     0.12.0
torch                     2.0.1+cu118
torchdiffeq               0.2.3
torchmetrics              1.0.1
torchsde                  0.2.5
torchvision               0.15.2+cu118
tqdm                      4.65.0
trampoline                0.1.2
transformers              4.25.1
triton                    2.0.0
typing_extensions         4.7.1
typing-inspect            0.9.0
tzdata                    2023.3
uc-micro-py               1.0.2
urllib3                   1.26.16
uvicorn                   0.23.1
wcwidth                   0.2.6
webencodings              0.5.1
websockets                11.0.3
Werkzeug                  2.3.6
wheel                     0.41.0
yacs                      0.1.8
yapf                      0.40.1
yarl                      1.9.2
zipp                      3.16.2

**Comment by Zhanghao0150** [comment]
$ python3
Python 3.10.12 (main, Jun 11 2023, 05:26:28) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import ssl
>>> print(ssl.OPENSSL_VERSION)
OpenSSL 3.0.2 15 Mar 2022
>>>

**Comment by Zhanghao0150** [comment]
大佬能改用aria2下载咩

**Comment by triphop** [comment]
please use another branch [bugfix/gradio_api_incompatibility](https://github.com/miaoshouai/miaoshouai-assistant/tree/bugfix/gradio_api_incompatibility) to see whether the issue above is gone.

```sh
cd <your_stable_diffusion_directory>/extensions/miaoshouai-assistant/
git checkout -b verfication origin/bugfix/gradio_api_incompatibility
git pull --rebase -p
```

change the 'disable_log_console_output" from **true** to **false** to switch on log output to console, so we can get detailed info.

```text
vim <your_stable_diffusion_directory>/extensions/miaoshouai-assistant/configs/settings.json

{
   ....
   "disable_log_console_output": false,
   ....
}
```

and after then, please **restart your webui again**

If the issue still be present, please give us the detailed log output to help us to address this issue.

**Comment by Zhanghao0150** [comment]
> 请使用另一个分支[错误修复/gradio_api_incompatibility](https://github.com/miaoshouai/miaoshouai-assistant/tree/bugfix/gradio_api_incompatibility)来查看上述问题是否消失。
> 
> ```shell
> cd <your_stable_diffusion_directory>/extensions/miaoshouai-assistant/
> git checkout -b verfication origin/bugfix/gradio_api_incompatibility
> git pull --rebase -p
> ```
> 
> 将“disable_log_console_output”从 **true** 更改为 **false**，以打开日志输出到控制台，以便我们可以获取详细信息。
> 
> ```
> vim <your_stable_diffusion_directory>/extensions/miaoshouai-assistant/configs/settings.json
> 
> {
>    ....
>    "disable_log_console_output": false,
>    ....
> }
> ```
> 
> 之后，请**再次重新启动您的 WebUI**
> 
> 如果问题仍然存在，请向我们提供详细的日志输出，以帮助我们解决此问题。

我之后重装python环境了，现在可以正常使用了，谢谢啦。我猜测之前的原因可能是那个问题是因为不支持OpenSSL 3.0.2版本导致的。

**Comment by triphop** [comment]
嗯, 的确是因为我们使用的一个package不支持OpenSSL 3.x导致的!
